src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1142,6 +1188,71 @@ protected virtual void ReportUnassigned(Symbol symbol, SyntaxNode node, int slot
 
             // mark the variable's slot so that we don't complain about the variable again
             _alreadyReported[slot] = true;
+
+            return;
+
+            bool needsImplicitFieldInitialization()
+                => _requireOutParamsAssigned
+                && CurrentSymbol is MethodSymbol { MethodKind: MethodKind.Constructor, ContainingType.TypeKind: TypeKind.Struct };
+
+            bool useWarningForStructThis(Symbol thisParameter, int thisSlot)
+            {
+                if (!needsImplicitFieldInitialization())
+                {
+                    return false;
+                }
+
+                Debug.Assert(_emptyStructTypeCache
+                    .GetStructInstanceFields(thisParameter.ContainingType)
+                    .Any(field => VariableSlot(field, thisSlot) is var fieldSlot && (fieldSlot == -1 || !State.IsAssigned(fieldSlot))));
+
+                foreach (var field in _emptyStructTypeCache.GetStructInstanceFields(thisParameter.ContainingType))
+                {
+                    if (_emptyStructTypeCache.IsEmptyStructType(field.Type))
+                        continue;
+
+                    if (field is TupleErrorFieldSymbol)
+                        continue;
+
+                    int slot = VariableSlot(field, thisSlot);
+                    if (slot == -1 || !State.IsAssigned(slot))
+                    {
+                        Debug.Assert(CurrentSymbol.EnclosingThisSymbol().Type.GetMembers(field.Name).Contains(field, ReferenceEqualityComparer.Instance));
```

Comment:
>Debug.Assert(CurrentSymbol.EnclosingThisSymbol().Type.GetMembers(field.Name).Contains(field, ReferenceEqualityComparer.Instance)) [](http://example.com/codeflow?start=24&length=129)

Can this assert be moved to the beginning of the `foreach` block?

Reply:
I thought a bit about this, and I think I'm going to delete the assert--it just verifies that the fields returned in 'GetStructInstanceFields' are in fact fields of the thing we asked for fields for.